Sources/SecureXPC/Server/XPCServer.swift

```diff
@@ -152,11 +174,21 @@ public class XPCServer {
     /// - Throws: If this route has already been registered.
     public func registerRoute<M: Decodable>(_ route: XPCRouteWithMessageWithoutReply<M>,
                                             handler: @escaping (M) throws -> Void) throws {
-        if self.routes.keys.contains(route.route) {
-            throw XPCError.routeAlreadyRegistered(route.route.pathComponents)
-        }
-        
-        self.routes[route.route] = ConstrainedXPCHandlerWithMessageWithoutReply(handler: handler)
+        try checkUniqueRoute(route: route.route)
+        self.routes[route.route] = ConstrainedXPCHandlerWithMessageWithoutReplySync(handler: handler)
+    }
+    
+    /// Registers a route that has a message and can't receive a reply.
```

Comment:
I would use more context-specific terminology:

> /// Registers a route that sends message and can't receive a reply.

This is similar to how below you say "expects a reply" rather than "has a reply"

Reply:
Hmm, this was my attempt to be specific. It's not that it doesn't expect a reply, it very much cannot have one - it's quite literally impossible with the way the code is written.